test/langtools/jdk/javadoc/tool/testTransformer/TestTransformer.java

```diff
@@ -0,0 +1,170 @@
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug      8298405
+ * @summary  Markdown support in the standard doclet
+ * @library  /tools/lib ../../lib
+ * @modules jdk.compiler/com.sun.tools.javac.api
+ *          jdk.compiler/com.sun.tools.javac.main
+ *          jdk.javadoc/jdk.javadoc.internal.tool
+ * @build    toolbox.ToolBox javadoc.tester.*
+ * @run main TestTransformer
+ */
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.ServiceLoader;
+import java.util.stream.StreamSupport;
+
+import com.sun.source.doctree.DocCommentTree;
+import com.sun.source.doctree.DocTree;
+import com.sun.source.doctree.RawTextTree;
+import com.sun.source.doctree.TextTree;
+import com.sun.source.util.DocTreeFactory;
+import com.sun.source.util.DocTrees;
+
+import javadoc.tester.JavadocTester;
+import jdk.javadoc.doclet.DocletEnvironment;
+import jdk.javadoc.doclet.StandardDoclet;
+import toolbox.ToolBox;
+
+public class TestTransformer extends JavadocTester {
+    public static void main(String... args) throws Exception {
+        var tester = new TestTransformer();
+        tester.runTests();
+    }
+
+    ToolBox tb = new ToolBox();
+
+    @Test
+    public void testFindStandardTransformer_raw() throws Exception {
+        int count = 0;
+        var sl = ServiceLoader.load(DocTrees.DocCommentTreeTransformer.class);
+        for (var t : sl) {
+            if (t.name().equals("standard")) {
+                out.println("Found " + t);
+                count++;
+            }
+        }
+        checking("transformer");
+        if (count == 1) {
+            passed("expected transformer found");
+        } else {
+            failed("expected transformer not found");
+        }
+    }
+
+    @Test
+    public void testFindStandardTransformer_stream() throws Exception {
+        var dct = getTransformer("standard");
+        checking("transformer");
+        if (dct.isPresent()) {
+            out.println("Found " + dct.get());
+            passed("expected transformer found");
+        } else {
+            failed("transformer not found");
+        }
+    }
+
+    private Optional<DocTrees.DocCommentTreeTransformer> getTransformer(String name) {
+        var sl = ServiceLoader.load(DocTrees.DocCommentTreeTransformer.class);
+        return StreamSupport.stream(sl.spliterator(), false)
+                .filter(p -> p.name().equals(name))
```

Comment:
ServiceLoader specification suggests another way of streaming:
```suggestion
        return sl.stream()
                .map(ServiceLoader.Provider::get)
                .filter(t -> t.name().equals(name))
```
Would it be the same and more readable?

Reply:
Hmm. It's longer, but arguably simpler to comprehend than using a spliterator. I've changed to code.